presentation/components/src/test/instance-filter-builder/PresentationInstanceFilterDialog.test.tsx

```diff
@@ -0,0 +1,185 @@
+} from "../../presentation-components/instance-filter-builder/PresentationInstanceFilterBuilder";
+import { stubRaf } from "./Common";
+import { PresentationInstanceFilterDialog } from "../../presentation-components";
+import { expect } from "chai";
+
+describe("PresentationInstanceFilterDialog", () => {
+  stubRaf();
+  const category = createTestCategoryDescription({ name: "root", label: "Root" });
+  const classInfo = createTestECClassInfo();
+  const propertiesField = createTestPropertiesContentField({
+    properties: [{ property: { classInfo, name: "prop1", type: "string" } }],
+    name: "prop1Field",
+    label: "propertiesField",
+    category,
+  });
+  const descriptor = createTestContentDescriptor({
+    selectClasses: [{ selectClassInfo: classInfo, isSelectPolymorphic: false }],
+    categories: [category],
+    fields: [propertiesField],
+  });
+  const initialFilter: PresentationInstanceFilterInfo = {
+    filter: {
+      operator: PropertyFilterRuleGroupOperator.And,
+      conditions: [{
+        field: propertiesField,
+        operator: PropertyFilterRuleOperator.IsNull,
+        value: undefined,
+      }],
+    },
+    usedClasses: [classInfo],
+  };
+
+  const imodelMock = moq.Mock.ofType<IModelConnection>();
+  const onCloseEvent = new BeEvent<() => void>();
+
+  beforeEach(() => {
+    async function* generator() {
+      return;
+    }
+    imodelMock.setup((x) => x.query(moq.It.isAnyString(), moq.It.isAny(), moq.It.isAny())).returns(() => generator());
+    imodelMock.setup((x) => x.key).returns(() => "test_imodel");
+    imodelMock.setup((x) => x.onClose).returns(() => onCloseEvent);
+    const metadataProvider = getIModelMetadataProvider(imodelMock.object);
+    sinon.stub(metadataProvider, "getECClassInfo").callsFake(async () => {
+      return new ECClassInfo(classInfo.id, classInfo.name, classInfo.label, new Set(), new Set());
+    });
+  });
+
+  afterEach(() => {
+    onCloseEvent.raiseEvent();
+    imodelMock.reset();
+  });
+
+  before(async () => {
+    await NoRenderApp.startup({
+      localization: new EmptyLocalization(),
+    });
+    await UiComponents.initialize(new EmptyLocalization());
+    await Presentation.initialize();
+    Element.prototype.scrollIntoView = sinon.stub();
+  });
+
+  after(async () => {
+    Presentation.terminate();
+    UiComponents.terminate();
+    await IModelApp.shutdown();
+    sinon.restore();
+  });
+
+  it("invokes 'onInstanceFilterApplied' with filter", async () => {
+    const spy = sinon.spy();
+    const { container, getByText, getByDisplayValue } = render(<PresentationInstanceFilterDialog
+      imodel={imodelMock.object}
+      descriptor={descriptor}
+      onInstanceFilterClosed={() => { }}
+      onInstanceFilterApplied={spy}
+      isOpen={true} />);
+
+    const applyButton = container.querySelector<HTMLInputElement>(".presentation-instance-filter-button-bar .iui-high-visibility:disabled");
+    expect(applyButton).to.not.be.null;
```

Comment:
This looke like unnecessary. Check below will faile if this is `null`. Same in other test